server/src/main.py

```diff
@@ -128,35 +122,30 @@ async def download_file(workspace_id: str, path: str):
             detail=f"404_NOT_FOUND: {path} not found in {workspace_id}",
         )
 
-
 @app.post("/workspaces/{workspace_id}/upload")
-async def upload_file(workspace_id: str, files: list[UploadFile]):
-    # TODO: Replace this with the logic to upload a file to the given workspace
-    # There may be multiple files in this so you will need to handle all of them
-    # -- for a later date --
-
+@app.post("/workspaces/{workspace_id}/upload/{directory_path:path}")
```

Comment:
For consistency with other endpoints, can we call this `path:path` instead of `directory_path:path`.